kinto_portier/views.py

```diff
@@ -39,6 +41,33 @@ def persist_nonce(request):
     return nonce
 
 
+def colander_querystring_validator(request, schema=None, deserializer=None, **kwargs):
+    """
+    Validate the querystring against the schema defined on the service.
+
+    The content of the querystring is deserialized, validated and stored in the
+    ``request.validated`` attribute.
+
+    .. note::
+
+        If no schema is defined, this validator does nothing.
+
+    :param request: Current request
+    :type request: :class:`~pyramid:pyramid.request.Request`
+
+    :param schema: The Colander schema
+    :param deserializer: Optional deserializer, defaults to
+        :func:`cornice.validators.extract_cstruct`
+    """
+    import colander
+
+    class RequestSchema(colander.MappingSchema):
+        querystring = schema
```

Comment:
No that's just because the schema is passed as a parameter to the validator so we don't have it before.
Also about the import you are right, it is not meant to be there.